tests/specs/opcodes/opcodes-v4-spec.k

```diff
@@ -0,0 +1,387 @@
+          </txnIndexMapGroup>
+        </txnIndexMap>
+    requires S <Int 1000
+
+// callsub target 	4 	heavy_check_mark 	1
+  claim <k> callsub LABEL => . </k>
+        <callStack> (.List => ListItem(CURRENT_PC +Int 1)) CS </callStack>
+        <pc> CURRENT_PC:Int => JUMP_PC </pc>
+        <labels> .Map[LABEL <- JUMP_PC] </labels>
+        <jumped> _ => true </jumped>
+    requires size(CS) <Int MAX_CALLSTACK_DEPTH
+
+// retsub 	4 	heavy_check_mark 	1
+  claim <k> retsub => . </k>
+        <callStack> (ListItem(RET_PC) => .List) _ </callStack>
+        <pc> _ => RET_PC </pc>
+        <jumped> _ => true </jumped>
+
+// shl 	4 	heavy_check_mark 	1
+  claim <k> shl => . </k>
+        <stack> 5 : 1 : XS => 32 : XS </stack>
+        <stacksize> S => S -Int 1 </stacksize>
+
+  claim <k> shl => . </k>
+        <stack> 1 : (2 ^Int 63) : XS => 0 : XS </stack>
+        <stacksize> S => S -Int 1 </stacksize>
+
+// shr 	4 	heavy_check_mark 	1
+  claim <k> shr => . </k>
+        <stack> 5 : 64 : XS => 2 : XS </stack>
+        <stacksize> S => S -Int 1 </stacksize>
+
+  claim <k> shr => . </k>
+        <stack> 3 : 2 : XS => 0 : XS </stack>
+        <stacksize> S => S -Int 1 </stacksize>
+
+// sqrt 	4 	heavy_check_mark 	4
+// Square root algorithm is currently very slow for verification.
+//   claim <k> sqrt => . </k>
+//         <stack> 15 : XS => 3 : XS </stack>
+
+// bitlen 	4 	heavy_check_mark 	1
+  claim <k> bitlen => . </k>
+        <stack> 8 : XS => 4 : XS </stack>
+
+  claim <k> bitlen => . </k>
+        <stack> 10 : XS => 4 : XS </stack>
+
+  claim <k> bitlen => . </k>
+        <stack> b"\x10" : XS => 5 : XS </stack>
+        
+// exp 	4 	heavy_check_mark 	1
+  claim <k> exp => . </k>
+        <stack> 5 : 3 : XS => 243 : XS </stack>
+        <stacksize> S => S -Int 1 </stacksize>
+
+  claim <k> exp => panic(INVALID_ARGUMENT) </k>
+        <stack> 0 : 0 : _ </stack>
+
+  claim <k> exp => panic(INT_OVERFLOW) </k>
+        <stack> 100 : 2 : _ </stack>
+
+// expw 	4 	heavy_check_mark 	1
+  claim <k> expw => . </k>
+        <stack> 5 : 3 : XS => 243 : 0 : XS </stack>
+
+  claim <k> expw => . </k>
+        <stack> 64 : 2 : XS => 0 : 1 : XS </stack>
+
+  claim <k> expw => panic(INVALID_ARGUMENT) </k>
+        <stack> 0 : 0 : _ </stack>
+
+  claim <k> expw => panic(INT_OVERFLOW) </k>
+        <stack> 150 : 2 : _ </stack>
+
+// b+ 	4 	heavy_check_mark 	10
+  claim <k> b+ => . </k>
+        <stack> b"\x03" : b"\x04" : XS => b"\x07" : XS </stack>
+        <stacksize> S => S -Int 1 </stacksize>
+
+  // Are byte operations supposed to be able to overflow?
+  claim <k> b+ => . </k>
+        <stack> b"\x01\x00\x00\x00\x00\x00\x00\x00\x00" : b"\x04" : XS 
+            =>  b"\x01\x00\x00\x00\x00\x00\x00\x00\x04": XS
+        </stack>
+        <stacksize> S => S -Int 1 </stacksize>
+
+  claim <k> b+ => panic(ILL_TYPED_STACK) </k>
+        <stack> b"1" : 123 : _ </stack>
+  
+// b- 	4 	heavy_check_mark 	10
```

Comment:
What do these comment lines mean @nwatson22 ?

Reply:
I copied a table from one of our documentation pages to serve as headers for the sections for each opcode. It doesn't have a meaning here except the first column which shows the opcode and the last which shows the cost. This is something that should be cleaned up, but we are planning on moving these tests into a markdown file anyway which will display this information more clearly.